packages/api/lambdas/sf-event-sqs-to-db-records/index.js

```diff
@@ -128,18 +130,26 @@ const handler = async (event) => {
   });
 
   const sqsMessages = get(event, 'Records', []);
+  const batchItemFailures = [];
 
-  return await Promise.all(sqsMessages.map(async (message) => {
+  await Promise.all(sqsMessages.map(async (message) => {
+    let cumulusMessage;
     const executionEvent = parseSQSMessageBody(message);
-    const cumulusMessage = await getCumulusMessageFromExecutionEvent(executionEvent);
-
+    try {
+      cumulusMessage = await getCumulusMessageFromExecutionEvent(executionEvent);
+    } catch (error) {
+      log.error(`Writing message failed on getting message from execution event: ${JSON.stringify(message)}`, error);
+      return batchItemFailures.push({ itemIdentifier: message.messageId });
+    }
     try {
       return await writeRecords({ ...event, cumulusMessage, knex });
     } catch (error) {
       log.error(`Writing message failed: ${JSON.stringify(message)}`, error);
       return sendSQSMessage(process.env.DeadLetterQueue, message);
     }
   }));
+
+  return { batchItemFailures };
```

Comment:
We have units covering this is what the function returns, *however* we don't have a test, in integration, that shows on `getCumulusMessageFromExecutionEvent` failure, the outputs map to the correct convention and the system does what we expect.     I'm fine with the results of our discussion in the Arch meeting re: performance testing being outside the scope of this ticket.